internal/configs/configurator.go

```diff
@@ -1085,23 +1087,21 @@ func (cnf *Configurator) UpdateConfig(cfgParams *ConfigParams, ingExes []*Ingres
 // As a result of the changes, the configuration for TransportServers is updated and some TransportServers
 // might be removed from NGINX.
 func (cnf *Configurator) UpdateGlobalConfiguration(globalConfiguration *conf_v1alpha1.GlobalConfiguration,
-	transportServerExes []*TransportServerEx) (updatedTransportServerExes []*TransportServerEx, deletedTransportServerExes []*TransportServerEx, err error) {
+	transportServerExes []*TransportServerEx) ([]*TransportServerEx, []*TransportServerEx, error) {
 	cnf.globalCfgParams = ParseGlobalConfiguration(globalConfiguration, cnf.staticCfgParams.TLSPassthrough)
 
+	updatedTransportServerExes := []*TransportServerEx{}
+	deletedTransportServerExes := []*TransportServerEx{}
+
 	for _, tsEx := range transportServerExes {
 		if cnf.CheckIfListenerExists(&tsEx.TransportServer.Spec.Listener) {
 			updatedTransportServerExes = append(updatedTransportServerExes, tsEx)
-
 			err := cnf.addOrUpdateTransportServer(tsEx)
```

Comment:
warnings from config generation need to be propagated back to the Controller

please note that this function is removed in https://github.com/nginxinc/kubernetes-ingress/pull/1415
a new one is created https://github.com/nginxinc/kubernetes-ingress/blob/8db9f1029fa4ab2ddd9c66ee8985f6c574616626/internal/configs/configurator.go#L1076

this function will also need to propagate warnings -- https://github.com/nginxinc/kubernetes-ingress/blob/8db9f1029fa4ab2ddd9c66ee8985f6c574616626/internal/configs/configurator.go#L634

Reply:
We confined Warnings to `generateTransportServerConfig()` as the supporting infrastructure isn't available. Warnings are handled at a higher level and the TransportServer status is updated. Waiting on https://github.com/nginxinc/kubernetes-ingress/pull/1415 for this.